client/lib/feed-stream-store/index.js

```diff
@@ -34,10 +25,10 @@ function siteKeyMaker( post ) {
 }
 
 function mixedKeyMaker( post ) {
-	if ( post.feed_ID && post.feed_item_ID ) {
+	if ( post.feed_ID && ( post.feed_item_ID || post.ID ) ) {
```

Comment:
this... may not work.

on blog streams, we tack in a feed_ID and the post has a ID from the blog ID / post ID pair.So the ID isn't a feed_item_id, it's a blog post ID..

Reply:
ah, good catch then

Reply:
is there any item returned back from the api that:

1. has a feed_id but does not have a feed_item_ID
2. isn't from wpcom (does not have a blog/postId)

Reply:
> has a feed_id but does not have a feed_item_ID

yup, see the read/sites/:site stream

> isn't from wpcom (does not have a blog/postId

this is the gross part. we fake a blog_ID and ID for pure external items, replacing them with the feed ID and feed_item_ID. You can determine when we do this by looking at the is_external property.

Reply:
i mean't anything that is both of those properties at once. aka something that would flop on the `mixedKeyMaker`

Reply:
We discussed this on Slack. If an item has both a feed_item_ID and a feed_ID, then that is always a valid pair to use.

Otherwise falling back to site_ID, ID is the safest bet.